libs/datastructures/include/hpx/datastructures/tuple.hpp

```diff
@@ -597,6 +608,38 @@ namespace hpx { namespace util {
     };
 #endif
 
+#if defined(HPX_HAVE_CXX17_STD_VARIANT)
+    template <std::size_t I, typename... Ts>
+    struct tuple_element<I, std::variant<Ts...>>
+    {
+        using type = typename util::at_index<I, Ts...>::type;
+
+        static constexpr HPX_HOST_DEVICE HPX_FORCEINLINE type& get(
+            std::variant<Ts...>& var) noexcept
+        {
+            return std::get<I>(var);
+        }
+
+        static constexpr HPX_HOST_DEVICE HPX_FORCEINLINE type const& get(
+            std::variant<Ts...> const& var) noexcept
+        {
+            return std::get<I>(var);
+        }
+
+        static constexpr HPX_HOST_DEVICE HPX_FORCEINLINE type&& get(
+            std::variant<Ts...>&& var) noexcept
+        {
+            return std::get<I>(std::move(var));
+        }
+
+        static constexpr HPX_HOST_DEVICE HPX_FORCEINLINE type const&& get(
+            std::variant<Ts...> const&& var) noexcept
+        {
+            return std::get<I>(std::move(var));
+        }
+    };
+#endif
```

Comment:
A variant is not a tuple like thing.

Reply:
...and still `std::get` has to work with it. The standard says:

> Index-based value accessor: If `v.index() == I`, returns a reference to the value stored in `v`. Otherwise, throws `std::bad_variant_access`. The call is ill-formed if `I` is not a valid index in the variant.

Reply:
As noted above, those are two orthogonal things.